### ISSUE TITLE
HCELEC-411 Error not thrown when there is special character in the metadata file name

### ISSUE THREAD

**Issue by KuzminaAnna** [GlobalPathogenAnalysisService/gpas-cli#51]
**Describe the problem/error:** When there are special characters within metadata file name, error is not throwing when uploading the sample. Tested different samples, upload is either success or stuck.

**Expected behavior:** Since EC 1.0.8,  directory/metadata file name only allows \./:_-, any other characters are not allowed and would error would throw.  Expect the upload to throw error instead of successful upload or stuck.

**Steps to Reproduce:**  

Upload a meatadata with special characters in the file name, such as "@`~?" through EC
Expect the validation to throw error "The only special characters allowed are \./:_-". Actual the error did not throw, the upload process is either successful or stuck.
**System Identification:** DEV
**Reproducible?** Yes
**If multiple areas are impacted list areas:** EC 1.0.9 integrated with gpas-cli 0.2.1

**Comment by KuzminaAnna** [comment]
![Screen Shot 2022-07-08 at 11 03 12 AM](https://user-images.githubusercontent.com/24322003/178029321-0c5f3e4d-a116-46d4-a100-95ef15a5d68f.png)

**Comment by natesun1** [comment]
Please see the following screenshot with gpas-cli 0.2.1
![Screen Shot 2022-07-08 at 11 24 08 AM](https://user-images.githubusercontent.com/102683639/178032151-b73ea6c1-8246-435e-84e0-e54e41ae01a8.png)

**Comment by bede** [comment]
I have added a pre-validation check that raises a ValidationError if the path fails a regex:
```python
    if not bool(re.match(r"^[A-Za-z0-9\\\s./_-]+$", str(upload_csv))):
        raise ValidationError(
            [{"error": "upload csv path contains illegal characters"}]
        )
```

**Comment by deepakcherian06** [comment]
Is this fixed in gpas-cli version 0.2.2 ?
